languages/swift/exercises/concept/higher-order-functions/.docs/introduction.md

```diff
@@ -0,0 +1,120 @@
+### Function types
+
+We have seen before how function signatures are defined using a function name followed by comma-separated list of parameter labels along with their types, followed by `->` and the type of the values returned by the function.
+
+If one were to remove all of the names and labels from this signature, they would be left with the type of the function.
+
+```swift
+func shoppingList(item: String, quantity: Int) -> String {
+  "You need to buy \(quantity) \(item)."
+}
+// shoppingList: (String, Int) -> String
+
+func repeater(_ phrase: String, times: Int) -> String {
+  var result = ""
+  for _ in 1...times {
+    result += phrase
+  }
+  return result
+}
+// repeater: (String, Int) -> String
+
+func printString(line: String) {
+  print(line)
+}
+// printString: (String) -> ()
+```
+
+Notice how the types of the first two functions are the same, `(String, Int) -> String`, even though the two functions work veery differently. They both take a `String` and an `Int` as input and return a `String`. And, as one can see from the third function's type, even though the function body doesn't appear to return anything, it actually implicitly returns `()`, the only value of type `Void`. Note that `()` and `Void` are interchangeable as type names, so the type of `printString` could also be written as `(String) -> Void`.
+
+These function types can be used in the same manner as any other type in Swift, for example, they can be used in the declaration of variables or constants. And since functions are values in Swift, already existing functions can be assigned to these constants/variables:
+
+```swift
+var stringAndIntToString: (String, Int) -> String
+
+stringAndIntToString = shoppingList
+stringAndIntToString("carrots", 3)
+// => "You need to buy 3 carrots."
+
+stringAndIntToString = repeater
+stringAndIntToString("carrots", 3)
+// => "carrotscarrotscarrots"
+```
+
+As with other variables/constants, the value you assign to it must be of the correct type:
+
+```swift
+stringAndIntToString = printShoppingList
+// Error: Cannot assign value of type '(String) -> ()' to type '(String, Int) -> String'
+```
+
+### Function types as parameter types
+
+As function types can be used anywhere other types can be used, it follows that they can be used as parameter types and passed into other functions and called from within those functions.
+
+```swift
+func distribute(over strings: [String],
+                andInt i: Int,
+                function f: (String, Int) -> String
+               ) -> [String] {
+  var result = [String]()
+  for string in strings {
+    result.append(f(string, i))
+  }
+  return result
+}
```

Comment:
Maybe make this example even more simple? E.g. by having less arguments and not using an array or something like that.

Reply:
Simplified the functions and removed the second version from the introduction.